OPAL/tac/src/main/scala/org/opalj/tac/fpcf/analyses/cg/AndroidICCAnalysis.scala

```diff
@@ -0,0 +1,1413 @@
+        clone.componentType = componentType
+        clone.receiver = receiver
+        clone
+    }
+
+    def evaluateUseSites(
+                          tacCode: AITACode[TACMethodParameter, ValueInformation],
+                          useSites: List[Int]
+    ): Unit = {
+        val statements = tacCode.stmts
+        useSites.foreach { ln ⇒
+            val stm = statements(ln)
+            if (stm.isVirtualMethodCall) {
+                val name = stm.asVirtualMethodCall.name
+
+                val parameter = stm.asVirtualMethodCall.params
+                if (name == addAction) {
+                    actions += parameter.head.asVar.value.toString.replaceAll(paramRegex, "")
+                }
+                if (name == addCategory) {
+                    categories += parameter.head.asVar.value.toString.replaceAll(paramRegex, "")
+
+                }
+                if (filterDataMethods.contains(name)) {
+                    name match {
+                        case `addDataAuthority` ⇒ dataAuthorities += (parameter.head.asVar.value.toString.
+                            replaceAll(paramRegex, "") + parameter.last.asVar.value.toString.
+                            replaceAll(paramRegex, ""))
+                        case `addDataPath` ⇒ dataPaths += (parameter.head.asVar.value.toString.
+                            replaceAll(paramRegex, "") + parameter.last.asVar.value.toString.
+                            replaceAll(paramRegex, ""))
+                        case `addDataScheme` ⇒ dataSchemes += parameter.head.asVar.value.toString.
+                            replaceAll(paramRegex, "")
+                        case `addDataSSP` ⇒ dataSSPs += parameter.head.asVar.value.toString.
+                            replaceAll(paramRegex, "")
+                        case `addDataType` ⇒ dataTypes += parameter.head.asVar.value.toString.
+                            replaceAll(paramRegex, "")
+                    }
+                }
+            }
+            if (stm.isAssignment) {
+                if (stm.asAssignment.expr.asVirtualFunctionCall.name == registerReceiver) {
+                    if (stm.asAssignment.expr.asVirtualFunctionCall.receiver.asVar.definedBy.head > -1)
+                        registeredReceivers += ObjectType(statements(stm.asAssignment.expr.asVirtualFunctionCall.
+                            receiver.asVar.definedBy.head).asAssignment.expr.asGetField.objRef.asVar.value.toString.
+                            replaceAll(receiverRegex, "").replaceAll("\\.", "/"))
+                }
+            }
+        }
+    }
+}
+
+class implicitIntent(var caller: Method) {
+
+    var action: String = ""
+    var categories: ListBuffer[String] = ListBuffer.empty[String]
+    var iType: String = ""
+    var iPackage: String = ""
+    var iData: String = ""
+    var componentTypes: ListBuffer[String] = ListBuffer.empty[String]
+    var calledMethods: ListBuffer[String] = ListBuffer.empty[String]
+    var pc: Int = 0
```

Comment:
All of this looks very strange. Should this maybe be a case class? Why are there setter methods for some fields but not for others?